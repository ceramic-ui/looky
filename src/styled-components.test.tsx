import * as React from "react";
import * as renderer from "react-test-renderer";
import styled from "styled-components";
import Looky from ".";

const theme = {
  media: ["576px", "768px", "992px", "1200px"],
  spacings: { small: "8px", default: "16px", large: "32px" }
};

const looky = Looky<typeof theme>(t => t.media);
const gutterX = looky(
  "gutterX",
  v => `
padding-left: ${v};
padding-right: ${v};
`
);
const Comp = styled<{ gutterX: string | Array<string | null> }, "div">("div")`
  ${gutterX};
`;

test("works with styled-components for single value", () => {
  expect(
    renderer
      .create(
        <Comp theme={theme} gutterX="spacings.small">
          Hello
        </Comp>
      )
      .toJSON()
  ).toMatchSnapshot();
});

test("works with styled-components for multiple value", () => {
  expect(
    renderer
      .create(
        <Comp
          theme={theme}
          gutterX={[
            "spacings.small",
            null,
            "spacings.default",
            "spacings.large"
          ]}
        >
          Hello
        </Comp>
      )
      .toJSON()
  ).toMatchSnapshot();
});
