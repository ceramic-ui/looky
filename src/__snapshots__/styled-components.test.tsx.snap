// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`works with styled-components for multiple value 1`] = `
.c0 {
  padding-left: 8px;
  padding-right: 8px;
}

@media (min-width:768px) {
  .c0 {
    padding-left: 16px;
    padding-right: 16px;
  }
}

@media (min-width:992px) {
  .c0 {
    padding-left: 32px;
    padding-right: 32px;
  }
}

<div
  className="c0"
>
  Hello
</div>
`;

exports[`works with styled-components for single value 1`] = `
.c0 {
  padding-left: 8px;
  padding-right: 8px;
}

<div
  className="c0"
>
  Hello
</div>
`;
